working/augmentation-libraries/feature-specification.md

```diff
@@ -142,59 +141,58 @@ It is a compile-time error if:
     is an *augmenting* declaration, described below). *This is the same error
     conceptually as having a name collision in one file.*
 
-*   An augmentation library contains any `part` directives.
+*   An augmentation contains any `part` directives.
 
-### Applying an augmentation library
+### Applying an augmentation
 
-A main library applies an augmentation to itself using a new import directive
-that looks like this:
+A main library applies an augmentation to itself using a new `augment` directive
+which looks like this:
 
 ```dart
-import augment 'some_augmentation.dart';
+augment 'some_augmentation.dart';
```

Comment:
This reads strangely to me now because in all the other places where we use `augment` it means that the subsequent thing is the thing *being augmented* but here it means *thing doing the augmenting*.

I don't have any great suggestions, but maybe keep `import augment` because that might be a little clearer? Perhaps `with augment`?

Reply:
Yes I agree, this is the one part I don't like now, for the same reasons. I suggested "augmentation" above for this but then its yet another special word to introduce.

Reply:
It would be nice to not have to introduce every conjugation of "augment" into the language: `augment`, `augments`, `augmentation`, `augmenting`, `augmentable`, `augmenter`, etc.. :)

I think `import augment` works OK.

Reply:
Fwiw, to me this is the strongest argument for just re-using the "part" directive 🤣. But, we would be breaking the semantics of it, so I think that is a no-go. But, we could just be bold...

Reply:
> Fwiw, to me this is the strongest argument for just re-using the "part" directive 🤣.

I would *love* to repurpose part files for this feature. "Augment" works OK as a word, but it's not exactly common and is pretty hard for ESL speakers from what I've gathered.

> But, we would be breaking the semantics of it, so I think that is a no-go. But, we could just be bold...

Yeah, it's probably a bridge too far. :( From a quick scrape of a pub package corpus:

```
-- File (56652 total) --
  45719 ( 80.701%): no part files    ===================================
   7759 ( 13.696%): is part file     ======
   3174 (  5.603%): uses part files  ===
```

So part files are actually pretty common. I suspect almost all of those are generated code?

Reply:
> So part files are actually pretty common. I suspect almost all of those are generated code?

Could you check how many of these files have multiple extensions, as a proxy for generated files (*.*.dart)?

Reply:
From skimming the file paths as they blew by on my terminal, I didn't notice a lot of extensions that suggested they were generated.

Reply:
We could use `part` here, and then whether the part file starts with `part of ....` or `augment library ...` decides whether it can have imports or not. Whether it's a simple part file or a library augmentation part file. (But if we allow `augment` in every file, #3576, we no longer have library augmentations, we just have files which may or may not include member augmentation. Then it's just about whether that file allows imports or it inherits the parent file's imports. At which point the inclusion syntax shouldn't mention `augment`, and I don't know which name would fit so it could be a `*word* part of ...` vs `part of ....`. )